docs/architecture/adr-002-lite-client.md

```diff
@@ -0,0 +1,439 @@
+passes, there is a fork, and evidence should be published so the validators get
+slashed. We leave the mechanics of evidence to a future document. For now, the
+lite client will just log an error and exit. If the verification fails, it
+means the backup that provided the conflict is bad and should be removed.
+
+#### Skipping Sync
+
+Skipping sync is essentially the same as sequential, except for a few points:
+
+- instead of verifying sequential headers, we attempt to "skip" ahead to the
+  full node's most recent height
+- skipping is only permitted if the validator set has not changed too much - ie.
+  if +1/3 of the last trusted validator set has signed the commit for the height we're attempting to skip to
+- if the validator set changes too much, we "bisect" the height space,
+  attempting to skip to a lower height, recursively. 
+- in the worst case, the bisection takes us to a sequential height
+
+### Requester
+
+The requester is simply a Tendermint RPC client. It makes requests to full
+nodes. It uses the `/commit` and `/validators` endpoints to get signed headers
+and validator sets for relevant heights. It may also use the `/status` endpoint
+to get the latest height of the full node (for skipping verification).
+
+### Detect
+
+The detection module is for checking if any of the backup nodes
+are reporting conflicting information. It requests headers from each backup node
+and compares them with a verified header from the primary. If there is a
+conflict, it attempts to verify the conflicting header via the verifier. If it
+can be verified, it indicates an attack on the lite clients that should be
+punishable. The relevant information (ie. the two conflicting commits) are
+passed to the publisher.
+
+### Publisher
+
+For now, the publisher just logs an error, write the conflicting commits to a
+file, and exits. We leave it to a future document to describe how this
+information can actually be published to the blockchain so the validators can be
+punished. Tendermint may need to expose a new RPC endpoint to facilitate this.
+
+### Address Book
+
+For now this is a simple list of HTTPS addresses corresponding to full nodes
+that the node connects to. One is randomly selected to be the primary, while
+others serve as backups. It's essential that the lite node connect to at least
+one correct full node in order to detect conflicts in a timely fashion. We keep
+this mechanism simple for now, but in the future a more advanced peer discovery
+mechanism may be utilized.
+
+### Verifier
+
+Most of the lite client logic resides in the verifier, where commits for a
+header are actually verified. In order to abstract over all the data in the
+Tendermint data structures, we use a set of traits that include only the
+information absolutely necessary for the lite client. From this perspective, we
+have the following traits.
+
+#### Header
+
+A Header has only a height, time, and validator sets, and can be identified by its hash:
```

Comment:
This looks way cleaner than before!

Doesn't it also need one hash for all state machine data?

Reply:
Yes but that's not relevant to the Tendermint side of the lite client protcol - this is really just about verifying updates to the validator set and the new headers they sign. once those updates are trusted, then you can look inside at app hashes to verify state queries

Reply:
Oh I see where I was confused, this is defining a trait not a struct. Perhaps we can change this sentence to the following:

```
A Header must contain a height, time, and validator sets. It can be uniquely identified by its hash:
```

(The core contribution being removing the word only)